### PR TITLE
Fix spelling: commited -> committed

### PR DESCRIPTION
**Pull request by ottok**
In most places in the code the variable names are spelled correctly as
COMMITTED but in a couple places not. This fixes them and ensures the
variable is always called COMMITTED everywhere.

### CODE REVIEW

**Comment by ottok** [comment]
BTW, please re-open https://github.com/facebook/rocksdb/issues/6276 as the issue itself is still unfixed just like it was then the issue was opened, there was no logical reason to close the issue.

File: include/rocksdb/utilities/transaction.h [hunk 1]
```diff
@@ -491,7 +491,7 @@ class Transaction {
     AWAITING_PREPARE = 1,
     PREPARED = 2,
     AWAITING_COMMIT = 3,
-    COMMITED = 4,
+    COMMITTED = 4,
```
Comment (pdillinger) [inline include/rocksdb/utilities/transaction.h]: This is a public API, so I think it's worth keeping the old name as an alias for the new name, which IIRC can be done with `COMMITED = COMMITTED, // old misspelled name`

File: java/src/main/java/org/rocksdb/Transaction.java [hunk 2]
```diff
@@ -1788,7 +1788,7 @@ public long getId() {
     AWAITING_PREPARE((byte)1),
     PREPARED((byte)2),
     AWAITING_COMMIT((byte)3),
-    COMMITED((byte)4),
+    COMMITTED((byte)4),
```
Comment (pdillinger) [inline java/src/main/java/org/rocksdb/Transaction.java]: This is also a public API, so let's do our best to keep the old name as an alias for the new name. It's not 100% possible for Java enums, but close enough, IMHO: https://stackoverflow.com/a/37092410/454544

File: include/rocksdb/utilities/transaction.h (hunk 1 above)
Comment (ottok) [reply include/rocksdb/utilities/transaction.h]: Added

File: java/src/main/java/org/rocksdb/Transaction.java (hunk 2 above)
Comment (ottok) [reply java/src/main/java/org/rocksdb/Transaction.java]: Added

**Comment by ottok** [comment]
Updated and force pushed new version with aliases for old COMMITED and a pending change in HISTORY.md.

**Comment by ottok** [comment]
.. and `allow edits from maintainers` is selected in case you want to further edit this in the same commit, but you can also iterate it more in later commits.

**Pull request closed without merging**